### PR TITLE
Port Group O test to pytest ref #3945

### PR DESCRIPTION
**Pull request by ShrillShrestha**
Refactor Group O mentioned in #3945

- test_random_invert
- test_random_posterize
- test_random_solarize
- test_random_adjust_sharpness
- test_random_autocontrast
- test_random_equalize

Please let me know if I need to change anything.

### CODE REVIEW

File: test/test_transforms.py [hunk 1]
```diff
@@ -1834,6 +1764,83 @@ def test_pad_with_mode_F_images(self):
         assert_equal(padded_img.size, [edge_size + 2 * pad for edge_size in img.size], check_stride=False)
 
 
+def _test_randomness(fn, trans, configs):
+    random_state = random.getstate()
+    random.seed(42)
+    img = transforms.ToPILImage()(torch.rand(3, 16, 18))
+
+    for p in [0.5, 0.7]:
+        for config in configs:
```
Comment (NicolasHug) [inline test/test_transforms.py:1773]: Looks like we don't even test on more than one config for a given function, so maybe we can remove this for loop, and change `[{}]` into `{}` (and related changes)

File: test/test_transforms.py [hunk 2]
```diff
@@ -1834,6 +1764,83 @@ def test_pad_with_mode_F_images(self):
         assert_equal(padded_img.size, [edge_size + 2 * pad for edge_size in img.size], check_stride=False)
 
 
+def _test_randomness(fn, trans, configs):
```
Comment (NicolasHug) [inline test/test_transforms.py:1767]: This is fine as-is for me. If we want, we can go one step further about parametrization here. I'm thinking of somthing like this:


```py
@pytest.mark.skipif(stats is None, reason="scipy.stats not available")
@pytest.mark.parametrize('fn, trans, conf', [
    (F.invert, transforms.RandomInvert, {}),
    (F.posterize, transforms.RandomPosterize, {"bits": 4}),
   ...

])
@pytest.mark.parametrize('p', (.5, .7))
def test_randomness(fn, trans, conf, p):
```

and we could get rid of all the other functions and the 2 for loops inside. WDYT?

Again this is optional: feel free to leave as-is if you prefer ;)

**Review by NicolasHug — COMMENTED** [review]
Thanks @ShrillShrestha ! This looks great, I made some minor comment, LMK what you think

File: test/test_transforms.py (hunk 2 above)
Comment (ShrillShrestha) [reply test/test_transforms.py]: Yeah, I think, we can parametrize it. I have made the changes.

**Comment by NicolasHug** [comment]
> +30 −70

Nice!

Thanks a lot @ShrillShrestha !

**Comment by ShrillShrestha** [comment]
@NicolasHug, thanks for guiding me. :)

**Pull request merged**